File: 594. Longest Harmonious Subsequence/solution.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        from collections import Counter
        count = Counter(nums)
        ans = 0
        for num in count:
            if num + 1 in count:
                ans = max(ans, count[num] + count[num + 1])
        return ans

File: 594. Longest Harmonious Subsequence/Solution.java
class Solution {
    public int findLHS(int[] nums) {
        Map<Integer, Integer> count = new HashMap<>();
        for (int num : nums) {
            count.put(num, count.getOrDefault(num, 0) + 1);
        }
        int ans = 0;
        for (Map.Entry<Integer,Integer> entry : count.entrySet()) {
            Integer next = count.get(entry.getKey() + 1);
            if (next == null) {
                continue;
            }
            ans = Math.max(ans, entry.getValue() + next);
        }
        return ans;
    }
}